allenact/algorithms/onpolicy_sync/engine.py

```diff
@@ -1495,28 +1519,40 @@ def backprop_step(
             for reduction, p in zip(reductions, all_params):
                 reduction.wait()
 
+        if hasattr(self.actor_critic, "compute_total_grad_norm"):
+            total_grad_norm = self.actor_critic.compute_total_grad_norm().item()
+        else:
+            total_grad_norm = 0.0
+
         nn.utils.clip_grad_norm_(
             self.actor_critic.parameters(),
             max_norm=max_grad_norm,  # type: ignore
         )
 
         self.optimizer.step()  # type: ignore
+        return total_grad_norm
 
     def _save_checkpoint_then_send_checkpoint_for_validation_and_update_last_save_counter(
         self, pipeline_stage_index: Optional[int] = None
     ):
+        model_path = None
         self.deterministic_seeds()
-        if self.worker_id == self.first_local_worker_id:
+        # if self.worker_id == self.first_local_worker_id:
```

Comment:
I guess this means we won't support having distributed training save a local copy of the model on each machine if the filesystem is not shared? It seems reasonable, since we have a new option, which I assume will usually enabled,  but just to be sure.

Reply:
yes! I don't think that saving checkpoints at every machine is necessary as well. Maybe we should have an option to turn it on?